src/core/network/injectSendNetworkRequest.js

```diff
@@ -39,7 +39,11 @@ export default ({ logger, networkStrategy, isRetryableHttpStatusCode }) => {
     }
 
     const executeRequest = (retriesAttempted = 0) => {
-      return networkStrategy(url, stringifiedPayload).then(response => {
+      return networkStrategy({
+        url,
+        body: stringifiedPayload,
+        documentMayUnload: payload.getDocumentMayUnload()
```

Comment:
This is the crux of why `sendBeacon` wasn't being used before. We weren't passing `documentMayUnload`.